tests/unit/Model/Command/InteractionTest.php

```diff
@@ -20,54 +22,192 @@ public function initTimeMock(): void
 
     /**
      * @test
-     * @dataProvider provideConstructorName
+     */
+    public function shouldRaiseExceptionWithInvalidAttributeName(): void
+    {
+        $this->expectException(DomainException::class);
+        $this->expectExceptionMessage(
+            'Value "invalid^*!@" does not match type identifier format requirement (must contain only of alphanumeric chars, dash or underscore)');
+        $command = Interaction::withItem('bookmarks', 'user-id', 'item-id');
+        $command->setAttribute('invalid^*!@', 'value');
```

Comment:
I suggest not repeating the whole exception message and assert only part of it, for better readability of the test.

Also, the `expectException` and `expectExceptionMessage` should be called right before the call which is expected to raise exception (to avoid false positives - if the exception is raised from different call).

Something like this:

```suggestion
        $command = Interaction::withItem('bookmarks', 'user-id', 'item-id');
        
        $this->expectException(DomainException::class);
        $this->expectExceptionMessage("invalid^*!@" does not match type identifier format requirement');
        $command->setAttribute('invalid^*!@', 'value');
```

And similarly in this whole test file.

Reply:
I didn't even know `expectExceptionMessage ` method worked like that. I don't like that but ok.